### DISCUSSION TITLE
Pocketbase JSVM, I need to delete the modules cache

### DISCUSSION THREAD

**Discussion by Fra146** [pocketbase/pocketbase#4220] [Q&A] [answered]
Using pocketbase as a framework with JS. I need to load a json file so I use `require('./file.json')`, however if I change the content of the file and re-run the function containing the require it will return the same object as before.

How do I force the JSVM to open the file again, effectively wiping the modules cache?

**Comment by dschissler** [comment]
You will need to place the file in your `__hooks` directory in order for Pocketbase to autorestart from watching the file..

**Reply by Fra146** [reply]
I really wouldn't want pocketbase to restart though. The file is changed by a script once every 4 minutes or so.

**Reply by dschissler** [reply]
If you're needs are different depending upon if you're in development mode or not, then you can make two ways to handle it.  For example, you could reread the file every time in development mode, but in production mode you could save the hassle of creating a cache for it by just using `require` and then waiting the "4 minutes or so" to restart it.

Pocketbase now has a `--dev` command line option that can be read off of the app struct.

**Comment by ganigeorgiev** [comment, marked as answer]
@Fra146 There is no option at the moment to clear the loaded modules.

An alternative of loading the json file with `require` is to load it with `$os.readFile` as you've attempted in https://github.com/pocketbase/pocketbase/discussions/4165.
The problem with this method is that it returns a byte slice and you'll have to convert it back to a string and then `JSON.parse` it to get back the js object. For example:

```js
const raw = $os.readFile(filepath)
const data = JSON.parse(String.fromCharCode(...raw))
```

I haven't benchmarked the above but if the file is very frequently accessed you can also optionally cache manually the read object in `$app.store().set("someKey", data)` (to retrieve it use the `$app.store().get("someKey")`).
